### PR TITLE
generic proxy: new codec callback methods to access connection

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message: generic proxy: new codec callback methods to access connection
Additional Description:

Tow new methods give the custom codec the full power to control the downstream/upstream connection.

Risk Level: low.
Testing: n/a.
Docs Changes: n/a.
Release Notes: n/a.
Platform Specific Features: n/a.

### CODE REVIEW

**Comment by RyanTheOptimist** [comment]
@soulxu friendly maintainer-on-call ping.

**Comment by phlax** [comment]
@soulxu this is still waiting on review

/wait-any

File: contrib/generic_proxy/filters/network/source/proxy.cc
Comment (soulxu) [inline contrib/generic_proxy/filters/network/source/proxy.cc:395]: Do we really need the `optional` prefix the method name? this `connect()` seems like just a helper method, we can rename it as other name

**Review by soulxu — COMMENTED** [review]
Sorry for the review late, recently busy on internal stuff.

File: contrib/generic_proxy/filters/network/source/proxy.cc
Comment (wbpcode) [reply contrib/generic_proxy/filters/network/source/proxy.cc:395]: sgtm.

Comment (wbpcode) [reply contrib/generic_proxy/filters/network/source/proxy.cc]: done.

**Review by soulxu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
